dvc/tree/local.py

```diff
@@ -325,7 +332,10 @@ def _upload(
             from_file, tmp_file, name=name, no_progress_bar=no_progress_bar
         )
 
-        self.protect(tmp_file)
+        if file_mode is not None:
+            self.chmod(tmp_file, file_mode)
+        else:
+            self.protect(tmp_file)
```

Comment:
It still looks weird to be calling `protect` from inside `tree._upload` but chmod is separated from protect now